sdk/src/logs/batch_log_processor.cc

```diff
@@ -334,6 +344,17 @@ bool BatchLogProcessor::Shutdown(std::chrono::microseconds timeout) noexcept
     synchronization_data_->cv.notify_one();
     worker_thread_.join();
   }
+#  ifdef ENABLE_ASYNC_EXPORT
+  // wait for running async exports <= 0
+  std::unique_lock<std::mutex> lock(synchronization_data_->async_export_waker_m);
+  synchronization_data_->async_export_waker.wait_for(lock, timeout, [this] {
+    std::lock_guard<std::mutex> lk(synchronization_data_->async_export_data_m);
+    return export_data_storage_->running_async_exports.size() <= 0;
+  });
+
+  while (CleanUpGarbageAsyncData() == false)
```

Comment:
Should we use async_export_waker to wait a event instead of busy wait here?

Reply:
Are you talking about async_export_waker.notify_all() to be called in other areas?

Reply:
I mean `while (CleanUpGarbageAsyncData() == false);` is a busy wait. `CleanUpGarbageAsyncData` only lock `async_export_data_m`, and this lock has a very small critial section, it's almost busy wait when worker thread is exited when exporter is still exporting data in other thread.